Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -909,17 +958,41 @@ variable {x y : V} -- TODO: rename to u, v, w instead?
 lemma cons_tail_eq (p : G.Walk x y) (hp : ¬ p.Nil) :
     cons (p.adj_snd hp) p.tail = p := by
   cases p with
-  | nil => simp only [nil_nil, not_true_eq_false] at hp
+  | nil => simp at hp
   | cons h q =>
     simp only [getVert_cons_succ, tail_cons_eq, cons_copy, copy_rfl_rfl]
 
+/--
+The reverse direction of `cons_tail_eq` for rewriting.
+-/
+lemma cons_tail_eq' (p : G.Walk x y) (hp : ¬ p.Nil) :
+    p = cons (p.adj_snd hp) p.tail := (cons_tail_eq _ hp).symm
```

Comment:
Sorry, I don't get why you need this lemma

Reply:
I don't recall why either, I'll remove it (and `concat_dropLast_eq'`)